bentoml/_internal/server/__init__.py

```diff
@@ -31,17 +32,45 @@
 
 @inject
 def ensure_prometheus_dir(
-    prometheus_multiproc_dir: str = Provide[
-        DeploymentContainer.prometheus_multiproc_dir
-    ],
+    directory: str = Provide[DeploymentContainer.prometheus_multiproc_dir],
```

Comment:
it seems to me the root cause of this issue might be we are using a fixed directory for all BentoServer on the same host: 

```
    prometheus_multiproc_dir = providers.Factory[str](
        os.path.join,
        bentoml_container.bentoml_home,
        "prometheus_multiproc_dir",
    )
```

Should we change it to have a `serve_id` or something similar in the file path? similar to the serve_id in the ServeInfo tracking event?

Reply:
yeah, I believe this is the best practice here

Reply:
Yeah it can be a solution. But it's a strange coupling that seems good, I think. The `prometheus_multiproc_dir` is just similar to a common pid file in Linux daemons. I think totally randomly generated in the `/var` or `/tmp` and passed to children is the best practice.